File: server-ts/src/core/service/account.ts
class AccountService {
  private repository: Repository<Account>;

  constructor(repository: Repository<Account>) {
    this.repository = repository;
  }

  get(address: string): Promise<Account> {
    return this.repository.get(address);
  }

  getAll(): Promise<Account[]> {
    return this.repository.getAll();
  }
}

export default AccountService;

File: server-go/src/core/service/account.go
package service

import (
	"server/src/core/entity"
	"server/src/core/repository"
)

type Account struct {
	repository repository.Account
}

func NewAccountService(repository repository.Account) *Account {
	return &Account{repository}
}

func (s *Account) GetAll() []entity.Account {
	return s.repository.GetAll()
}

func (s *Account) Get(address string) (entity.Account, error) {
	return s.repository.Get(address)
}
